crates/re_data_ui/src/lib.rs

```diff
@@ -189,3 +195,31 @@ pub fn annotations(
     annotation_map.load(ctx, query, std::iter::once(entity_path));
     annotation_map.find(entity_path)
 }
+
+// ---------------------------------------------------------------------------
+
+/// Build an egui table and configure it for the given verbosity.
+pub fn table_for_verbosity(
+    verbosity: UiVerbosity,
+    ui: &mut egui::Ui,
+) -> egui_extras::TableBuilder<'_> {
+    let table = egui_extras::TableBuilder::new(ui);
+    match verbosity {
+        UiVerbosity::Small | UiVerbosity::Reduced => {
+            // Be as small as possible in the hover tooltips. No scrolling related configuration, as
+            // the content itself must be limited (scrolling is not possible in tooltips).
+            table.auto_shrink([true, true])
+        }
+        UiVerbosity::MultiSelectionPanel => {
+            // Don't take too much vertical space to leave room for other selected items.
+            table
+                .auto_shrink([false, true])
+                .vscroll(true)
+                .max_scroll_height(100.0)
+        }
+        UiVerbosity::SelectionPanel => {
+            // We're alone in the selection panel. Let the outer ScrollArea do the work.
+            table.auto_shrink([false, true]).vscroll(false)
```

Comment:
I suggest we only disable scrolling if the number of elements in the table is small enough, to avoid the performance problem in https://github.com/rerun-io/rerun/issues/4426

Reply:
As per discussion yesterday, let's try to address that at `egui::Table` level. If that fail, I'll do another PR with hard-coded limits. I've tagged this issue with 0.12 so we dont forget:
- https://github.com/rerun-io/rerun/issues/4426